pkg/controller/kubefedcluster/controller_test.go

```diff
@@ -65,17 +65,17 @@ func TestThresholdCheckedClusterStatus(t *testing.T) {
 		"ClusterNotReadyButWithinFailureThreshold": {
 			clusterStatus: clusterStatus(corev1.ConditionFalse, t3, t3),
 			storedClusterData: &ClusterData{
-				clusterStatus: clusterStatus(corev1.ConditionTrue, t2, t1),
-				resultRun:     2},
-			expectedClusterStatus: clusterStatus(corev1.ConditionTrue, t3, t1),
-			expectedResultRun:     3,
+				clusterStatus: clusterStatus(corev1.ConditionFalse, t2, t1),
+				resultRun:     1},
+			expectedClusterStatus: clusterStatus(corev1.ConditionFalse, t3, t1),
+			expectedResultRun:     2,
 		},
 		"ClusterNotReadyAndCrossedFailureThreshold": {
 			clusterStatus: clusterStatus(corev1.ConditionFalse, t4, t4),
 			storedClusterData: &ClusterData{
 				clusterStatus: clusterStatus(corev1.ConditionTrue, t3, t1),
 				resultRun:     3},
-			expectedClusterStatus: clusterStatus(corev1.ConditionFalse, t4, t4),
+			expectedClusterStatus: clusterStatus(corev1.ConditionTrue, t4, t1),
```

Comment:
Here also you are changing a valid test. Extending to previous test and it explaination above. When the check is done 4th time (with the failureThreshold as 3) The cluster state changes from `ready=true` to `ready=false`. Also the transition time is also updated to t4 because that is the time of the cluster state transition.